CRM/Sepa/BAO/SEPAMandate.php

```diff
@@ -37,6 +37,10 @@ class CRM_Sepa_BAO_SEPAMandate extends CRM_Sepa_DAO_SEPAMandate implements HookI
    * @throws \CRM_Core_Exception
    */
   public static function self_hook_civicrm_pre(PreEvent $event): void {
+    if ('delete' === $event->action) {
```

Comment:
The old method got called for `create`, `save` and `update` actions (see [here](https://github.com/civicrm/civicrm-core/blob/86f1d61adf587e7c8fbcf8289afec6175ceb86de/Civi/Api4/Generic/Traits/DAOActionTrait.php#L167)), so I think we should whitelist instead of blacklist possible actions for this hook, which include e.g. `view` as well, see [here](https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_pre/#parameters).